url8867e6a4e2_cyberpython_JAwesomeChart_tgz-pJ8-com_gmigdos_jawesomechart_renderers_twodimensional_horizontal_ColumnChartRendererJ8/src/com/gmigdos/jawesomechart/renderers/twodimensional/horizontal/Basic2DHorizontalChartRenderer.java

```diff
@@ -0,0 +1,359 @@
+        boolean drawValueAxis = isDrawValueAxisOn();
+        boolean drawHorizontalLines = isDrawHorizontalLinesOn();
+        if (drawValueAxis || drawHorizontalLines) {
+            DecimalFormat df = getDecimalFormat();
+            double minData = series.getMinValue();
+            double maxData = series.getMaxValue();
+                    
+            double margin = drawValueAxis ? getValueAxisMargin() : 0;
+            double axisMarkerSize = drawValueAxis ? getAxisMarkerSize() : 0;
+            double dataDistance = getDataDistance(maxData, minData);
+            double paddingTop = getPaddingTop();
+            double paddingBottom = getPaddingBottom();
+            double height = context.getHeight() - paddingTop - paddingBottom - getOffsetForLabelAxis(context, labels);
+            double width = context.getWidth() - getOffsetForValueAxis(context, series, valueAxisMarks);
+            double posH = getPositiveAreaSize(series, height);
+            double x;
+            double y;
+            double minValue;
+            double maxValue;
+            Font font = getValueAxisFont();
+            Color valueAxisTextColor = getValueAxisTextColor();
+            Color axisMarkerColor = getAxisMarkerColor();
+            Color axisMarkerColorForZero = getAxisMarkerColorForZero();
+            Color horizontalLineColor = getHorizontalLineColor();
+            Color horizontalLineColorForZero = getHorizontalLineColorForZero();
+            Stroke axisMarkerStroke = getAxisMarkerStroke();
+            Stroke axisMarkerStrokeForZero = getAxisMarkerStrokeForZero();
+            Stroke horizontalLineStroke = getHorizontalLineStroke();
+            Stroke horizontalLineStrokeForZero = getHorizontalLineStrokeForZero();
+
+            context.setFont(font);
+            double maxValueWidth = drawValueAxis ? Utilities.calculateWidestValueWidth(context, valueAxisMarks, df) : 0;
+
+
+            if (maxData < 0) {
+                maxValue = 0;
+                minValue = minData;
+            } else if (minData > 0) {
+                maxValue = maxData;
+                minValue = 0;
+            } else {
+                maxValue = maxData;
+                minValue = minData;
+            }
+
+            context.save();
+            context.translate(maxValueWidth, paddingTop + posH);
+
+            for (Double value : valueAxisMarks) {
+
+                if (value >= minValue && value <= maxValue) {
+                    y = -(value * height / dataDistance);
+
+                    if (value != 0) {
+                        if (drawHorizontalLines) {
+                            context.setPaint(horizontalLineColor);
+                            context.setStroke(horizontalLineStroke);
+                            x = margin + axisMarkerSize;
+                            context.draw(new Line2D.Double(x, y, x + width, y));
+                        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 222:227, 235:240.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.